### ISSUE TITLE
Fidelity gate false positive: Contract V1 compares synthesized lambda-cache ordinals literally

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#3503]
## Summary

`FidelityGateTests.NoNewFidelityDiffsBeyondKnownDocket` reports
`StatementBodyLambdaInsideIf` as a new `OperandDiff` on current `main`
(`e1b3cac9`). **It is not a decompiler defect.** The recompiled IL is
behaviourally identical to the original; the gate is comparing
compiler-synthesized closure-cache *ordinals* literally.

Surfaced while triaging #3326, alongside #3490 and #3502 — but unlike those
two, this row should be fixed in the gate, not in the decompiler.

## The diff is ordinal-only

The opcode sequences are **byte-for-byte identical**:

```
original  : ldarg brfalse ldsfld dup brtrue pop ldsfld ldftn newobj dup stsfld ret ldnull ret
recompiled: ldarg brfalse ldsfld dup brtrue pop ldsfld ldftn newobj dup stsfld ret ldnull ret
```

The whole reported difference is three operands:

```
- <>c::<>9__103_0                          + <>c::<>9__128_0
- <>c::<StatementBodyLambdaInsideIf>b__103_0   + <>c::<StatementBodyLambdaInsideIf>b__128_0
- <>c::<>9__103_0  (stsfld)                + <>c::<>9__128_0  (stsfld)
```

Both names denote the same cache slot for the same lambda in the same closure
class. `ldsfld`/`ldftn`/`stsfld` all move together, so there is no
mismatched-slot possibility — only a renumbering.

## Why the number differs

`<>9__N_M` is Roslyn's lambda cache field: `N` is the containing method's
**declaration ordinal within the compilation unit**, `M` the lambda index
within that method. The ordinal is a property of the *unit being compiled*, not
of the method.

Confirmed against the real assembly with a metadata probe:

- `CfgSampleClass` declares **687** methods.
- `StatementBodyLambdaInsideIf` sits at metadata method-table index **97**.
- The synthesized member in the shipped assembly is
  `<>c::<StatementBodyLambdaInsideIf>b__103_0` — ordinal **103**.

So the ordinal is neither the metadata index nor anything the decompiler emits;
it is assigned by Roslyn from declaration order. The compile-back path
recompiles a **reconstructed skeleton**, whose member ordering and member count
are the harness's, not the original source file's — so Roslyn assigns **128**
for the same method. Nothing about the decompiled C# changed.

The corollary is that this row was passing on a coincidence: as long as the two
orderings happened to agree at this method, the operands matched. Growth in
`CfgSampleClass` (687 methods and rising, since fixtures are added
continuously) broke the coincidence. Expect more rows to flip the same way,
each looking like a fresh regression.

## Why the gate does not filter it

`FidelityCheck.ContractV1BodyDiffNormalization` (`FidelityCheck.cs:44`) is:

```csharp
IlBodyDiffNormalization.NormalizeVariableLayout
| IlBodyDiffNormalization.NormalizeCurrentAssemblyScope
| IlBodyDiffNormalization.NormalizePlatformAssemblyScope
```

Contract V1 already accepts that some IL differences are non-semantic — local
slot layout and assembly-scope spelling are both normalized away for exactly
this reason. Synthesized-member ordinals belong in the same category and were
simply never added.

## Suggested fix

Normalize the ordinal inside Roslyn-synthesized member names when comparing
operands — i.e. treat `<>9__N_M` and `<Name>b__N_M` as equal when `Name`
and `M` match, regardless of `N`.

Deliberately narrow, to avoid masking real defects:

- Normalize **only** the containing-method ordinal `N`. Keep `Name` and the
  per-method lambda index `M` significant, so a lambda bound to the wrong
  cache slot, or to a lambda of a different method, still fails.
- Do not normalize synthesized **type** names or arity.
- Do not extend it to user-authored names.

## Harness-boundary note

Per `AGENTS.md` this is a legitimate oracle concern rather than harness
compensation for a product gap: the product's decompiled C# is correct, and the
divergence is introduced by Roslyn when recompiling a differently-ordered unit.
That is the same category as the two normalizations Contract V1 already
performs. It is **not** a case of the harness covering for missing decompiler
behavior.

## Reproduction

```
dotnet build dotnet-inspect.slnx -c Release
dotnet run --project src\ILInspector.Decompiler.Tests -c Release --no-build -- \
  -class "ILInspector.Decompiler.Tests.FidelityGateTests"
```

`StatementBodyLambdaInsideIf` appears in the unexpected list. Observed on
Windows; not Windows-specific (`Speed=Slow`, so excluded from the PR lane by
`ci.yml:301` on Linux too).

## Do not pin this into `KnownDiffs`

Adding the row to the docket would record a decompiler defect that does not
exist, and would not stop the next method from flipping for the same reason.
